plat/ti/k3/common/k3_bl31_setup.c

```diff
@@ -99,12 +99,28 @@ void bl31_early_platform_setup2(u_register_t arg0, u_register_t arg1,
 
 void bl31_plat_arch_setup(void)
 {
-	arm_setup_page_tables(BL31_BASE,
-			      BL31_END - BL31_BASE,
-			      BL_CODE_BASE,
-			      BL_CODE_END,
-			      BL_RO_DATA_BASE,
-			      BL_RO_DATA_END);
+#define K3_MAP_BL31_TOTAL	MAP_REGION_FLAT(			\
```

Comment:
This does not look good to me, why make definitions just to then expand them only once in the very next line? Just add them to the array directly, or since the second parameter already takes a variable number of mappings you can add these to that array and drop this first parameter.

Reply:
I think this is just a matter of preference. The macros were defined above to make it clear what the regions are. In the arm platform code there are some common regions as well that are defined in arm_defs.h so the macros are only defined here for the special cases for that image. If you would prefer to remove the macro definitions and add them directly to the array for your platform I am happy to rework this for you. 

These mappings cannot be added to the second array as they have different types. The array in the second parameter can be generated statically however since the first array requires values only resolved during linking then it must be dynamically allocated.

Reply:
IMHO good coding style is more than just personal preference. Anyway, yes, it would be nice to have a simple array like we do in [ti/k3/common/k3_bl31_setup.c](https://github.com/ARM-software/arm-trusted-firmware/blob/master/plat/ti/k3/common/k3_bl31_setup.c#L19). Then just use comments to make it clear what the regions are.